internal/hardware/validator.go

```diff
@@ -175,19 +175,27 @@ func (v *validator) ListEligibleDisks(inventory *models.Inventory) []*models.Dis
 
 	// Sorting list by size increase
 	sort.Slice(eligibleDisks, func(i, j int) bool {
+		// 1. First, choose non-NVMe drives before NVMe drives
 		isNvme1 := isNvme(eligibleDisks[i].Name)
 		isNvme2 := isNvme(eligibleDisks[j].Name)
+		// If i is NVMe and j isn't, return false (j first)
+		// If j is NVMe and i isn't, return true (i first)
 		if isNvme1 != isNvme2 {
 			return isNvme2
 		}
 
-		// HDD is before SSD
-		switch v := strings.Compare(string(eligibleDisks[i].DriveType), string(eligibleDisks[j].DriveType)); v {
-		case 0:
-			return eligibleDisks[i].SizeBytes < eligibleDisks[j].SizeBytes
-		default:
+		// 2. Sort DriveType (HDD is before SSD)
+		if v := strings.Compare(string(eligibleDisks[i].DriveType), string(eligibleDisks[j].DriveType)); v != 0 {
 			return v < 0
 		}
+
+		// 3. Sort according to disk size
+		if eligibleDisks[i].SizeBytes != eligibleDisks[j].SizeBytes {
+			return eligibleDisks[i].SizeBytes < eligibleDisks[j].SizeBytes
+		}
+
+		// 4. Sort according to HCTL (physical order)
```

Comment:
It would be nice if this comment went into detail about why we're doing this, in case we want make changes to this in the future

Reply:
Done, and added explanations to the other steps as well.